### PR TITLE
add basic validation for studio preview for reverification block

### PR DESCRIPTION
**Pull request by zubair-arbi**
ECOM-1438

@awais786 @aamir-khan @wedaly 
Added `validate` method for `Reverification XBlock`.

### CODE REVIEW

**Comment by zubair-arbi** [comment]
screen shot 2015-04-29 at 5 28 58 pm
@wedaly @cahrens @frrrances I was able to use default validation classes of `XBlock` to add custom validation messages for `Reverification XBlock`. There is slight difference between how `Library Content Module` display's its message (warning/error) and this block will display its message (above screenshot).
Is this view good (validation messages in form of bullet points)?

File: edx_reverification_block/xblock/reverification_block.py [hunk 1]
```diff
@@ -249,3 +251,23 @@ def _render_template(template_path, context):
         template_str = ReverificationBlock._resource(template_path)
         template = Template(template_str)
         return template.render(Context(context))
+
+    def validate(self):
+        """
+        Basic validation for this XBlock.
+        """
+        reverifcation_block_validation = super(ReverificationBlock, self).validate()
```
Comment (wedaly) [inline edx_reverification_block/xblock/reverification_block.py]: Slight typo here: `reverifcation_block_validation` => `reverification_block_validation`

File: edx_reverification_block/xblock/reverification_block.py [hunk 2]
```diff
@@ -249,3 +251,23 @@ def _render_template(template_path, context):
...
+        if not isinstance(reverifcation_block_validation, Validation):
+            reverifcation_block_validation = Validation.copy(reverifcation_block_validation)
+
+        if self.is_configured:
```
Comment (wedaly) [inline edx_reverification_block/xblock/reverification_block.py]: Should this be `not self.is_configured`?

File: edx_reverification_block/xblock/reverification_block.py [hunk 3]
```diff
@@ -249,3 +251,23 @@ def _render_template(template_path, context):
...
+            reverifcation_block_validation.add(
+                ValidationMessage(
+                    ValidationMessage.WARNING,
+                    _(
```
Comment (wedaly) [inline edx_reverification_block/xblock/reverification_block.py]: For i18n in XBlocks, we need to use the `i18n` service.  This is how we used it in ORA2:
https://github.com/edx/edx-ora2/blob/master/openassessment/xblock/openassessmentblock.py#L456-L459

**Comment by wedaly** [comment]
Please add test cases for the `validate()` method (one where the XBlock is configured and one in which it isn't).

**Comment by frrrances** [comment]
@zubair-arbi From a UX standpoint, I think the bullet point listing is fine.

**Comment by cahrens** [comment]
@zubair-arbi The reason the library content block's validation message looks different is that they are using the StudioValidation subclasses. That allows you to control the summary message (and not show the bulleted messages below), as well as adding action links.

If you guys are happy with the more limited display that the default validation framework supports, then that is fine and you don't have to worry about moving the StudioValidation classes into xblock or xblock-utils.

File: edx_reverification_block/xblock/reverification_block.py [hunk 4]
```diff
@@ -249,3 +249,28 @@ def _render_template(template_path, context):
...
+    def _(self):
+        i18nService = self.runtime.service(self, 'i18n')
+        return i18nService.ugettext
+
+    def validate(self):
+        """
+        Basic validation for this XBlock.
+        """
+        reverification_block_validation = super(ReverificationBlock, self).validate()
+        if not isinstance(reverification_block_validation, Validation):
```
Comment (cahrens) [inline edx_reverification_block/xblock/reverification_block.py]: I don't think this copy is necessary.

**Comment by zubair-arbi** [comment]
@wedaly @cahrens I have added test for validate method. Please review.

**Comment by cahrens** [comment]
:+1:

**Comment by wedaly** [comment]
:+1:

**Pull request merged**